addons/mass_mailing_event/models/mailing_mailing.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api, _
+from odoo.exceptions import UserError
+
+
+class MassMailingCampaign(models.Model):
+    _inherit = "mailing.mailing"
+
+    @api.constrains('mailing_model_id', 'mailing_domain')
+    def _check_domain_for_event_registration(self):
+        for mass_mailing in self:
+            if mass_mailing.mailing_model_name == 'event.registration' and 'event_id' not in mass_mailing.mailing_domain:
+                raise UserError(_('You must filter on the event to contact attendees'))
```

Comment:
This makes no sense. Why would I be blocked if I try to mail all registrations of belgian people since beginning of the year, whatever the event ?

Reply:
@tde-banana-odoo 
This code is useful if "newbies" miss-click and contact all attendees (remark written in the pad ;) )
But yes, maybe can we "ignore" newbie, and remove the constrains...

Reply:
Preventing people from working is rarely a good idea. What is important is that coming from event (Mail Attendees button) domain should be good by default (event, not cancelled, ...).